### DISCUSSION TITLE
Velocity gradient in cartesian basis

### DISCUSSION THREAD

**Discussion by TRPrasanna** [loganoz/horses3d#214] [Q&A] [answered]
Is the variable named u_y that is output by horses2plt the derivative of x-velocity with respect to the cartesian y-coordinate? Based on what I read in the source code, this should be the case, but I am not observing this in the Tecplot output file that I am generating. I get a different value compared to the value that I compute in Tecplot using
```
{dudy}=ddY({u})
```

I have also tried calculating the surface integral of du/dy through a surface monitor (this is a part of what I want to achieve in the end) using, say something like:
```
         case ( DUDY)
      
!           **********************************
!           Computes the surface integral
!              val = \int du/dy dS
!           **********************************

            associate( Q_x => f % storage(1) % U_x, &
               Q_y => f % storage(1) % U_y, &
               Q_z => f % storage(1) % U_z, &
               Q => f % storage(1) % Q   )
!
            do j = 0, f % Nf(2) ;    do i = 0, f % Nf(1)
!
!              Compute the integral
!              --------------------

               invRho  = 1._RP / Q(IRHO,i,j)
               invRho2 = invRho * invRho
            
               uDivRho = [Q(IRHOU,i,j) , Q(IRHOV,i,j) , Q(IRHOW,i,j) ] * invRho2

               val = val + (invRho * Q_y(IRHOU,i,j) - uDivRho(1) * Q_y(IRHO,i,j)) &
                * f % geom % jacobian(i,j) * spAxi % w(i) * spAeta % w(j) 
            end do          ;    end do
            end associate
```

and I notice again that this wouldn't yield the expected value. Am I misunderstanding something here? Thank you for your time.

Edit: I forgot to mention some details. I am using the following line when using horses2plt
```
dimensionless = .true.
```
and my reference length is 1 meters (so the grid is the same in dimensional and non-dimensional coordinates), so I do not think that the issue concerns non-dimensionalization.

**Comment by TRPrasanna** [comment, marked as answer]
Upon further investigation, I found out what was happening. It was this following setting in the control file that was causing the unexpected values:

```
                      Gradient variables = Entropy
```
I guess the entropy derivatives were being stored in U_x, U_y and U_z when the aforementioned option was being used. Commenting this out fixed my issue.

**Comment by loganoz** [comment]
I am glad to check the problem is solved!
